src/main/java/Duke.java

```diff
@@ -1,10 +1,273 @@
+                otherArgs = splitInput[3];
+            }
+            else {
+                otherArgs = null;
+            }
+
+            params.put("task", name);
+            params.put("done", done);
+
+            switch(type) {
+            case "T":
+                params.put("type", "todo");
+                break;
+            case "D":
+                params.put("type", "deadline");
+                params.put("by", otherArgs);
+                break;
+            case "E":
+                params.put("type", "event");
+                params.put("at", otherArgs);
+                break;
+            default:
+                break;
+            }
+
+            taskMgr.addTask(params);
+        }
+    }
+
+    private static void updateTaskFile() {
+        String fileDataToWrite = taskMgr.toFileString();
+
+        try {
+            checkAndSetData();
+            Path dataFilePath = Paths.get(rootDirectory, dataLocation, dataFilename);
+
+            FileWriter fw = new FileWriter(dataFilePath.toString());
+            fw.write(fileDataToWrite);
+            fw.close();
+        }catch(IOException e){
+            System.out.println("Cannot update task file.");
+        }
+    }
+
+    private static void doTask(Command command) {
+    	String indexParam = command.getParam("index");
+        int index;
+        
+        try {
+            index = Integer.parseInt(indexParam);
+        } catch(NumberFormatException e) {
+        	System.out.println("Index is not a number");
+        	return;
+        }
+
+        try {
+            Task completedTask = taskMgr.doTask(index);
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.printf("%s\n", completedTask);
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private static void removeTask(Command command) {
+        String indexParam = command.getParam("index");
+        int index;
+
+        try {
+            index = Integer.parseInt(indexParam);
+        } catch(NumberFormatException e) {
+            System.out.println("Index is not a number");
+            return;
+        }
+
+        try {
+            Task completedTask = taskMgr.removeTask(index);
+            System.out.println("Noted I've removed this task:");
+            System.out.printf("%s\n", completedTask);
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+    
+    private static void addTask(Command command) {
+    	Task newTask;
+        try{
+            newTask = taskMgr.addTask(command.getParams());
+        }catch(IllegalArgumentException e){
+            System.out.println(e.getMessage());
+            return;
+        }
+
+        System.out.println("Got it. I've added this task: ");
+        System.out.printf("%s\n", newTask);
+
+        int numOfTasks = taskMgr.getTasklistLength();
+        String plural = (numOfTasks <= 1) ? "task" : "tasks";
```

Comment:
Suggest to rename "plural" variable to something more general, or more verbose. If the string is singular, it contradicts the variable name.